libcontainer/nsenter/nsexec.c

```diff
@@ -1210,7 +1227,10 @@ void nsexec(void)
 				}
 
 				/* Become root in the namespace proper. */
-				if (setresuid(0, 0, 0) < 0)
+				write_log(DEBUG, "Become root in the namespace proper.");
+				write_log(DEBUG, "uid: %d\n", config.uid);
+				// TODO get it from config
+				if (setresuid(config.uid, config.uid, config.uid) < 0)
```

Comment:
In particular I'm not sure this will work -- and even if it did work you can't use `config.uid` here for containers that have a root user because in those cases you definitely wouldn't be able to configure the container properly. Maybe we don't need to setuid for user namespaces before we unshare the other namespaces? I'm not sure.

Reply:
You right about ambient capabilities. I decided to put it into separate PR.
https://github.com/AlexeyPerevalov/runc/commit/f1f54ad9905f22dcb912d914ccf48168aa405730
Not yet ready - since for me it looks hacky.
The main intention here was also be able to run runc as non root user.

For containers that have a root user - do you mean container where we're running process as root or where FS has a root in files ownership? If the latest one - yes, for testing I "repacked" images and rewrote perm's of the files, images which are just pulled from public domain - are not suitable, unless some shifting fs is not applied (personally I didn't try)